packages/flutter/lib/src/widgets/localizations.dart

```diff
@@ -383,7 +391,6 @@ class _LocalizationsState extends State<Localizations> {
   T resourcesFor<T>(Type type) {
     assert(type != null);
     final T resources = _typeToResources[type];
-    assert(resources.runtimeType == type);
```

Comment:
you want to assert that the type matches, though. Probably using [`TypeMatcher`](https://master-docs-flutter-io.firebaseapp.com/flutter/widgets/TypeMatcher-class.html).